Sprint_4

Elena-Mirionkova opened the pull request.

Commit:
modified:   README.md
	modified:   tests.py

DolgoarshinnykhLG commented on .vscode/settings.json:
Необходимо исправить: эта директория лишняя в проекте. Необходимо убрать её из ветки и добавить в .gitignore в корне проекта

DolgoarshinnykhLG commented on __pycache__/main.cpython-311.pyc:
Необходимо исправить: эта директория лишняя в проекте. Необходимо убрать её из ветки и добавить в .gitignore в корне проекта

DolgoarshinnykhLG commented on tests.py:
```diff
@@ -18,7 +19,109 @@ def test_add_new_book_add_two_books(self):
+        collector.add_new_book(name)
+        if 0 < length < 41:
+            assert len(collector.books_genre) == 1
+        else:
```
Нужно исправить: использование условных блоков в теле теста лишает их атомарности. Можно попробовать, например, расширить количество параметров или их переработать дабы избежать этой ситуации. Если не получается объединить их все, то отдельные можно разместить отдельно исключив из параметризации

Commit:
new file:   .gitignore

Commit:
deleted:    .vscode/settings.json
	deleted:    __pycache__/main.cpython-311.pyc
	deleted:    __pycache__/main.cpython-38.pyc
	deleted:    __pycache__/test.cpython-38-pytest-7.1.2.pyc
	deleted:    __pycache__/tests.cpython-311-pytest-8.4.1.pyc

Commit:
modified:   README.md
	modified:   tests.py
